modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/agg/Accumulators.java

```diff
@@ -161,6 +161,9 @@
         @Override public void add(Object... args) {
             assert args.length == 1 : args.length;
 
+            if (holder != null)
```

Comment:
What about queries like `SELECT NAME FROM products WHERE CAT_ID = (SELECT null UNION ALL SELECT 1)`? I think we should fail on such queries too.